modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module

```diff
@@ -8,66 +8,79 @@
 declare(strict_types=1);
 
 use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Entity\ContentEntityInterface;
 use Drupal\Core\Form\FormStateInterface;
-use Drupal\node\NodeInterface;
 use Drupal\oe_editorial_corporate_workflow\EntityStateTransitionBatch;
+use Drupal\workflows\Entity\Workflow;
 
 /**
  * Implements hook_menu_local_tasks_alter().
  */
 function oe_editorial_corporate_workflow_menu_local_tasks_alter(&$data): void {
-  $node = \Drupal::routeMatch()->getParameter('node');
-  if (!$node instanceof NodeInterface) {
-    return;
+  /** @var \Drupal\workflows\WorkflowInterface $workflow */
+  $workflow = Workflow::load('oe_corporate_workflow');
+  $entity_types = $workflow->get('type_settings')['entity_types'];
+  $entity = NULL;
+  foreach (array_keys($entity_types) as $entity_type) {
+    $route_entity = \Drupal::routeMatch()->getParameter($entity_type);
+    if (!$route_entity instanceof ContentEntityInterface) {
+      continue;
+    }
+
+    if (!in_array($route_entity->bundle(), $entity_types[$entity_type])) {
+      continue;
+    }
+
+    $entity = $route_entity;
```

Comment:
We can stop the cycle here, no need to iterate through all once we found the entity in the parameter.